helix-lock/src/main/java/org/apache/helix/lock/helix/ZKDistributedNonblockingLock.java

```diff
@@ -125,6 +218,79 @@ public void close() {
     _baseDataAccessor.close();
   }
 
+  @Override
+  public void handleDataChange(String dataPath, Object data) throws Exception {
```

Comment:
Can we leverage this handleDataChange to notify the waiting thread? Then we can use some Java internal lock.

Reply:
Yeah, we are using this one to trigger a onAcquiredLockNotification callback, which will notify the waiting thread.